fix(compiler): incorrectly inferring content type of SVG-specific title tag

crisbeto opened the pull request:
The parser has a list of tag definitions that it uses when parsing the template. Each tag has a `contentType` which tells the parser what kind of content the tag should contain. The problem is that the browser has two separate `title` tags (`HTMLTitleElement` and `SVGTitleElement`) and each of them has to have a different `contentType`, otherwise the parser will throw an error further down the pipeline.

These changes update the tag definitions so that each tag name can have multiple content types associated with it and the correct one can be returned based on the element's prefix.

Fixes #31503.

JoostK reviewed (commented):
WDYT about adding some tests to `packages/compiler/test/ml_parser/lexer_spec.ts`, e.g. testing the simple case but also testing with child elements?

crisbeto commented:
Sure. By "testing with child elements", do you mean having a `title` with children or `title` being the child of something specific?

JoostK commented:
A `title` with children. I believe that's allowed ([from what I read on MDN](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title)), and `PARSABLE_DATA` should indeed accept it.

crisbeto commented:
The extra unit tests have been added.

crisbeto committed:
fix(compiler): incorrectly inferring content type of SVG-specific title tag

The parser has a list of tag definitions that it uses when parsing the template. Each tag has a
`contentType` which tells the parser what kind of content the tag should contain. The problem is
that the browser has two separate `title` tags (`HTMLTitleElement` and `SVGTitleElement`) and each
of them has to have a different `contentType`, otherwise the parser will throw an error further down
the pipeline.

These changes update the tag definitions so that each tag name can have multiple content types
associated with it and the correct one can be returned based on the element's prefix.

Fixes #31503.

crisbeto commented:
Merge assistance: needs a presubmit.

atscott commented:
[presubmit](http://test/OCL:351188341:BASE:351188373:1610390877832:f3f5dc2f)
